### ISSUE TITLE
Release 10 dropped support for range queries on the MessageReader

### ISSUE THREAD

**Issue by zwalsh-toast** [twilio/twilio-java#782]
### Issue Summary

Release 10 appears to have dropped support for a feature of the API that exists in the documentation. The `MessageReader` class no longer has the methods `setDateSentBefore` or `setDateSentAfter`, in lieu of just `setDateSent`. This means that I can't query within a single day or across days with the SDK.

https://github.com/twilio/twilio-java/compare/9.0.0..10.0.0#diff-3fd79a945865becabb8b305f841be94f67fbac40df790e96b88e8fe300a25fe6L83-L90

### Code Snippet

Before:

```java
    public MessageReader setDateSentBefore(final ZonedDateTime dateSentBefore){
        this.dateSentBefore = dateSentBefore;
        return this;
    }
    public MessageReader setDateSentAfter(final ZonedDateTime dateSentAfter){
        this.dateSentAfter = dateSentAfter;
        return this;
    }
```

Now:

```java
    public MessageReader setDateSent(final ZonedDateTime dateSent) {
        this.dateSent = dateSent;
        return this;
    }
```

This feature still exists in code snippets on the [REST API documentation](https://www.twilio.com/docs/messaging/api/message-resource#list-messages-that-were-sent-before-a-specific-date) and the params still seem to be respected by the API, so I think this is a regression. Can this be fixed?

### Technical details:
* twilio-java version: 10.0+
* java version: any

**Comment by sbansla** [comment]
@zwalsh-toast Range fields(dateSentBefore) are something internal. It was a bug in previous SDK versions.
We will talk with internal team members to come up with conclusion on this.
Either remove this from docs or add this to SDK.

**Comment by zwalsh-toast** [comment]
@sbansla Thanks! For my use case, this feature is really useful, so I hope it can be supported.

**Comment by sbansla** [comment]
@zwalsh-toast We have released 10.1.2 with the fix.
